astropy/coordinates/representation/geodetic.py

```diff
@@ -62,12 +98,21 @@ def __init_subclass__(cls, **kwargs):
             raise AttributeError(
                 f"{cls.__name__} requires '_ellipsoid' or '_equatorial_radius' and '_flattening'."
             )
+        if not hasattr(
+            cls._wrap_angle, "unit"
+        ) or not cls._wrap_angle.unit.is_equivalent(u.deg):
+            raise u.UnitTypeError("Attribute _wrap_angle requires deg units.")
         super().__init_subclass__(**kwargs)
 
     def __init__(self, lon, lat=None, height=None, copy=True):
         if height is None and not isinstance(lon, self.__class__):
             height = 0 << u.m
 
+        if not lon.unit.is_equivalent(u.deg) or not lat.unit.is_equivalent(u.deg):
+            raise u.UnitTypeError(
+                f"{self.__class__.__name__} requires lat and lon with units of angle."
+            )
+        lon = Longitude(lon << u.deg, wrap_angle=self._wrap_angle)
```

Comment:
Why convert the input to degrees? 

Note that I'm not sure this is necessary (see larger comment). But if we stick with using `_wrap_angle`, we need to think a bit about how we can potentially make it faster - right now, the wrapping is done twice (I think).